static/app/components/smartSearchBar/index.tsx

```diff
@@ -932,10 +1151,62 @@ class SmartSearchBar extends React.Component<Props, State> {
 
       return;
     }
-
     const cursor = this.getCursorPosition();
     const query = this.state.query;
 
+    const {organization} = this.props;
+    const AST = this.state.parsedQuery;
+    if (AST && organization.features.includes('search-syntax-highlight')) {
+      const filterNode = this.getCursorToken(AST, cursor);
+      if (filterNode && filterNode.type === Token.Filter) {
+        let clauseStart: null | number = null;
+        let clauseEnd: null | number = null;
+        let replaceToken = replaceText;
+        if (item.type === 'tag-operator') {
```

Comment:
Also I think if we move this to a function we could early return in this conditionals and make this a bit easier to read